import Post from "../../models/post.model.js";
import User from "../../models/user.model.js";

const getFollowingPost = async (req, res) => {
  try {
    const userId = req.user._id;
    const user = await User.findById(userId);
    if (!user) return res.status(404).json({ error: "User not found" });

    const following = user.following;

    const feedPosts = await Post.find({ user: { $in: following } })
      .sort({ createdAt: -1 })
      .populate({
        path: "user",
        select: "-password",
      })
      .populate({
        path: "comments.user",
        select: "-password",
      });

    res.status(200).json(feedPosts);
  } catch (error) {
    console.log(`Error in the getFollowingPost controller ${error}`);
    res.status(500).json({ error: "Internal server error" });
  }
};

export default getFollowingPost;
